src/excited_workflow/source_datasets/land_cover.py

```diff
@@ -1,19 +1,117 @@
+    return np.array([np.datetime64(el) for el in data.to_numpy()])
+
+
+def dims_check(dat: xr.Dataset, target: xr.Dataset, var: str) -> bool:
+    """Check dimensions of two grids are the same.
+
+    Args:
+        dat: Preprocessed dataset
+        target: Target grid dataset
+        var: variable name to compare
+
+    Returns:
+        boolean of where arrays are the same.
+    """
+    return np.allclose(dat[var].to_numpy(), target[var].to_numpy())
+
+
 class LandCover(DataSource):
     """Copernicus land cover dataset."""
 
     name: str = "copernicus_landcover"
     variable_names: list[str] = ["lccs_class"]
 
+    def regrid(
+        self,
+        file: Path,
+        target_grid: xr.Dataset,
+        variables: list[str] | None = None,
+    ) -> xr.Dataset:
+        """Regrids one file to target dataset.
+
+        Args:
+            file: file to regrid
+            variables: List of variable names which should be downloaded.
+            target_grid: Grid to which the data should be regridded to.
+
+        Returns:
+            Regridded Dataset.
+        """
+        ds = xr.open_dataset(file, chunks={"lat": 2000, "lon": 2000})
+
+        # Set time to middle of bounds.
+        time_coords = _cftime_to_datetime(ds["time_bounds"].mean(dim="bounds"))
+        ds = ds.drop("time_bounds")
+        ds["time"] = time_coords
+
+        ds = ds[["lccs_class"]]  # Only take the class variable.
+        ds = ds.sortby(["lat", "lon"])
+        ds = ds.rename({"lat": "latitude", "lon": "longitude"})
+
+        if variables is not None:
+            ds = ds[variables]
+
+        ds = ds.regrid.most_common(target_grid, time_dim="time")
+
+        return ds
+
+    def preprocess(
+        self,
+        process_path: Path,
+        target_grid: xr.Dataset,
+        variables: list[str] | None = None,
+    ) -> None:
+        """Preprocess variable.
+
+        Args:
+            process_path: Output path for preprocessed files.
+            variables: List of variable names which should be downloaded.
+            target_grid: Grid to which the data should be regridded to.
+        """
+        self.validate_variables(variables)
+
+        files = list(self.get_path().glob("*.nc"))
+        if len(files) == 0:
+            msg = f"No netCDF files found at path '{self.get_path()}'"
+            raise FileNotFoundError(msg)
+
+        for file in files:
+            name = process_path / (file.stem + ".nc")
+
+            if name.is_file() and name.stat().st_size > 0:
+                dat = xr.open_dataset(name)
+                if not (
+                    dims_check(dat, target_grid, "latitude")
+                    and dims_check(dat, target_grid, "longitude")
+                ):
+                    dat.close()
+                    ds = self.regrid(file, target_grid, variables)
+                    ds.to_netcdf(name)
+            else:
+                ds = self.regrid(file, target_grid, variables)
+                ds.to_netcdf(name)
```

Comment:
I think you could refactor this to:

```suggestion
            if not name.is_file() or not valid_regrid_file(name, target_grid): 
                self.regrid(file, target_grid, variables).to_netcdf(name)
```
Where the validity check (dims, st_size) can be wrapped in a function such as `valid_regrid_file`

Reply:
Additionally, I could be nice to print something here so the user gets feedback during the long pre-processing.
E.g. 
```py
print(f"File '{name}' was not found or not valid for this target dataset. Regridding raw data to `target_grid`")
```

Reply:
Ok created a new function and adjusted this, ruff was unhappy with the long text though so I shortened it. If you think longer is clearer, I can change it back.